### PR TITLE
Requirements and Dockerfile

### PR DESCRIPTION
**Pull request by Anko59**
`docker-compose run aiden_app`

### CODE REVIEW

**Comment by Anko59** [comment]
I am struggling with the Dockerfile, I don't know much about it. This version seems to work but when I access `localhost:8000` I get Connection refused error

**Comment by Smirkey** [comment]
Idk if we need a docker image already, the best would be first to be able to run it locally then we can figure out a docker image ?

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+version: '1'
```
Comment (Smirkey) [inline docker-compose.yml]: this is old

**Comment by Smirkey** [comment]
@Anko59 I've pushed some changes, stuff to be considered
- we use poetry now (version 1.8.2)
- I've added a reverse proxy layer to docker compose with traefik, the url to use is `http://aiden.dev.localhost/`
- Included some changes in dockerfile + code to use selenium within docker containers

I've created a `Makefile` to store commands. The `run` command launches the app and `build` rebuilds the docker image (necessary for now to run changes made locally)

If it can be useful here are my docker & docker compose version

```console
➜  ~ docker --version        
Docker version 26.1.1, build 4cf5afa
➜  ~ docker compose version  
Docker Compose version v2.27.0
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,26 +1,14 @@
...
-    rev: 4.0.1
-    hooks:
-      - id: flake8
-        args: [
-          --max-line-length=140,
-          --ignore=E203 E266 E712 E902 W503 W504,
-          --per-file-ignores=aiden_app/agents/prompts.py:E501
-        ]
+      # Run the linter.
+      - id: ruff
```
Comment (Anko59) [inline .pre-commit-config.yaml:8]: This config looks very short to me. Don't we need to specify max line length for example?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+ENV POETRY_NO_INTERACTION=1 \
+    POETRY_VIRTUALENVS_IN_PROJECT=1 \
+    POETRY_VIRTUALENVS_CREATE=1 \
+    POETRY_CACHE_DIR=/tmp/poetry_cache
+
+WORKDIR /app
+
+# Install necessary packages for selenium
+RUN apt-get update && apt-get install -y \
```
Comment (Anko59) [inline Dockerfile:12]: image

**Review by Anko59 — COMMENTED** [review]
Great work Buillaume 
image

File: .pre-commit-config.yaml (hunk 2 above)
Comment (Smirkey) [reply .pre-commit-config.yaml:8]: it's specified within the `ruff.toml` file
It's just the basic config by ruff, we use it at my company and works well

**Pull request merged**